cocotb_build_libs.py

```diff
@@ -167,7 +172,9 @@ def _get_python_lib():
 # TODO [gh-1372]: make this work for MSVC which has a different flag syntax
 _base_warns = ["-Wall", "-Wextra", "-Wcast-qual", "-Wwrite-strings", "-Wconversion"]
 _ccx_warns = _base_warns + ["-Wnon-virtual-dtor", "-Woverloaded-virtual"]
-_extra_cxx_compile_args = ["-std=c++11"] + _ccx_warns
+_extra_cxx_compile_args = ["-std=c++11", "-fvisibility=hidden", "-fvisibility-inlines-hidden"] + _ccx_warns
+if os.name != "nt":
+    _extra_cxx_compile_args += ["-flto"]
```

Comment:
Why is this both here and on line 140?

Reply:
On line 140 are linker arguments and on line 177 are compiler arguments `-flto` needs to be passed to linker and compiler.